### ISSUE TITLE
"Couldn't retrieve identity from Git repository." if no global signing key is set.

### ISSUE THREAD

**Issue by seltzered** [prydonius/karn#31]
repo.go in https://github.com/prydonius/karn/pull/25 (implement gpg signing) added a getconfig for a signingkey (see [repo.go diff](https://github.com/prydonius/karn/pull/25/commits/4ebd74d74d1ae9cf0115553e2d28d8fe8bc5f8a5#diff-65e557a13cfc70927d6259fdf0113fff370ecddebb867153d67ef9d0877d8db4)). This causes a "Couldn't retrieve identity from Git repository" issue if a user hasn't setup a signingkey at all in their global ~/.gitconfig.

Suggest making this a soft-check, possibly not just for signingkey but for all global gitconfig identity attributes if the intent of karn is to not depend on any global identity aspects. 

Workaround for now is to set a global signing key in ~/.gitconfig before using karn.

**Comment by prydonius** [comment]
Good catch, it shouldn't error on this. Should be a quick patch, @seltzered let me know if you want to submit a fix!

**Comment by sidx1024** [comment]
I'm getting the same issue when name and email are not set in ~/.gitconfig
